torch/nn/utils/spectral_norm.py

```diff
@@ -3,20 +3,26 @@
 """
 import torch
 from torch.nn.functional import normalize
+from typing import Any, Optional, TypeVar
+from ..modules import Module
 
 
-class SpectralNorm(object):
+class SpectralNorm:
     # Invariant before and after each forward call:
     #   u = normalize(W @ v)
     # NB: At initialization, this invariant is not enforced
 
-    _version = 1
+    _version: int = 1
     # At version 1:
     #   made  `W` not a buffer,
     #   added `v` as a buffer, and
     #   made eval mode use `W = u @ W_orig @ v` rather than the stored `W`.
+    name: str
+    dim: int
+    n_power_iterations: int
+    eps: float
```

Comment:
These seem unnecessary, `mypy` is happy with only the annotations of `__init__()` as far as I can tell. Adding these is harmless, but we want to make a habit of this?

Reply:
This PR just copies existing annotations.
I thought this is done to annotate class members, which are likely to be accessed outside of the class scope.

Reply:
It's covered in PEP 526; several methods are valid it looks like. One could also leave these out, or use
```
self.name: str = name
```
inline instead. I don't have a strong preference either way.